common/browser_context.go

```diff
@@ -340,43 +341,24 @@ func (b *BrowserContext) Unroute(url goja.Value, handler goja.Callable) {
 }
 
 // WaitForEvent waits for event.
-func (b *BrowserContext) WaitForEvent(event string, optsOrPredicate goja.Value) any {
-	// TODO: This public API needs Promise support (as return value) to be useful in JS!
+func (b *BrowserContext) WaitForEvent(event string, optsOrPredicate goja.Value) (any, error) {
 	b.logger.Debugf("BrowserContext:WaitForEvent", "bctxid:%v event:%q", b.id, event)
 
-	var (
-		isCallable  bool
-		predicateFn goja.Callable
-		// TODO: Find out whether * time.Second is necessary.
-		timeout = b.browser.browserOpts.Timeout * time.Second //nolint:durationcheck
+	parsedOpts := NewWaitForEventOptions(
+		b.timeoutSettings.timeout(),
 	)
-	if gojaValueExists(optsOrPredicate) {
-		switch optsOrPredicate.ExportType() {
-		case reflect.TypeOf(goja.Object{}):
-			opts := optsOrPredicate.ToObject(b.vu.Runtime())
-			for _, k := range opts.Keys() {
-				switch k {
-				case "predicate":
-					predicateFn, isCallable = goja.AssertFunction(opts.Get(k))
-					if !isCallable {
-						k6ext.Panic(b.ctx, "predicate function is not callable")
-					}
-				case "timeout":
-					timeout = time.Duration(opts.Get(k).ToInteger()) * time.Millisecond
-				}
-			}
-		default:
-			predicateFn, isCallable = goja.AssertFunction(optsOrPredicate)
-			if !isCallable {
-				k6ext.Panic(b.ctx, "predicate function is not callable")
-			}
-		}
+	if err := parsedOpts.Parse(b.ctx, optsOrPredicate); err != nil {
+		return nil, fmt.Errorf("parsing waitForEvent options: %w", err)
 	}
 
-	return b.waitForEvent(event, predicateFn, timeout)
+	return b.waitForEvent(event, parsedOpts.PredicateFn, parsedOpts.Timeout)
 }
 
-func (b *BrowserContext) waitForEvent(event string, predicateFn goja.Callable, timeout time.Duration) any {
+func (b *BrowserContext) waitForEvent(event string, predicateFn goja.Callable, timeout time.Duration) (any, error) {
+	if event != "page" {
+		return nil, fmt.Errorf("\"page\" is the only event that is supported, you passed in %q", event)
```

Comment:
```suggestion
		return nil, fmt.Errorf(`incorrect event: %q, "page" is the only event supported", event)		
```

It's okay, but should this `"page"` be a constant somewhere, as it is used in multiple places.

Reply:
Yeah, that makes sense. Resolved in 4c096de323dcee4e6b88cc80cd8ea227fd8aef14 and d741d451341620082cb433f50a75f30710c0afbb.

Reply:
You create [this type](https://github.com/grafana/xk6-browser/commit/4c096de323dcee4e6b88cc80cd8ea227fd8aef14#diff-688b96a8694c1d64dd8e32b212658c9953b9045c1225b0cea4ec2796bf6f2f26R36) because we'll add more event types, right? Makes sense 👍

A small suggestion for [this](https://github.com/grafana/xk6-browser/commit/d741d451341620082cb433f50a75f30710c0afbb#diff-143d56aaa3500065e9f275f9d202a6182d95a4887685bd54f9861356f0e38230R724):

```diff
- wantErr: "incorrect event \"browser\", \"page\" is the only event supported",
+ wantErr: `incorrect event "browser", "page" is the only event supported`,
```

Reply:
Yes correct, there will be (might be) more types that we handle later on.

Suggestion added in c07abdf5286789f69e0308a53d4bc83fbc48510b.